Reqnroll/EnvironmentAccess/BuildMetadataProvider.cs

```diff
@@ -0,0 +1,267 @@
+﻿using Reqnroll.CommonModels;
+
+namespace Reqnroll.EnvironmentAccess
+{
+    public class BuildMetadataProvider(IEnvironmentInfoProvider environmentInfoProvider, IEnvironmentWrapper environment) : IBuilldMetadataProvider
+    {
+        private string GetVariable(string variable)
+        {
+            var varValue = environment.GetEnvironmentVariable(variable);
+            if (varValue is ISuccess<string>)
```

Comment:
With `if (varValue is ISuccess<string> success)` , we save a cast (more compile time safe)